pkg/netpol/eval/exposure.go

```diff
@@ -30,18 +31,20 @@ func generateNewPodName(index int) string {
 }
 
 // generateRepresentativePeers : generates and adds to policy engine representative peers where each peer
-// has namespace and pod labels inferred from single policy rule labels in the given list of selectors;
-// for example, if a rule within policy has namespaceSelector "foo: managed", then a representative pod in such a
+// has namespace and pod labels and requirements inferred from single entry of selectors in a policy rule list;
+//
+// - for example, if a rule within policy has an entry: namespaceSelector "foo: managed", then a representative pod in such a
 // namespace with those labels will be added, representing all potential pods in such a namespace.
-// generated representative peers are unique; i.e. if different rules (e.g in different policies or different directions) has same labels :
-// one representative peer is generated to represent both
-func (pe *PolicyEngine) generateRepresentativePeers(selectorsLabels []k8s.SingleRuleLabels, policyName, policyNs string) (err error) {
-	for i := range selectorsLabels {
+// - generated representative peers are unique; i.e. if different rules (e.g in different policies or different directions)
+// has same labels, one representative peer is generated to represent both
+func (pe *PolicyEngine) generateRepresentativePeers(selectors []k8s.SingleRuleSelectors, policyName, policyNs string) (err error) {
+	for i := range selectors {
 		// if ns labels of the rule selector was nil, then the namespace of the pod is same as the policy's namespace
-		if selectorsLabels[i].PolicyNsFlag {
-			_, err = pe.AddPodByNameAndNamespace(generateNewPodName(i), policyNs, &selectorsLabels[i])
+		if selectors[i].PolicyNsFlag {
+			_, err = pe.AddPodByNameAndNamespace(generateNewPodName(i), policyNs, &selectors[i])
```

Comment:
consider renaming `AddPodByNameAndNamespace` ? split to the various cases as different functions?

Reply:
done.
1. reverted all changes to `AddPodByNameAndNamespace ` (to the main branch version) so it applies only for ingress-controller cases (renaming this func, if neede - should be done in the ingress-analysis branch, not here)
2. added a new function  `addRepresentativePod` for our use-case (exposure-analysis).